docs/reference/aggregations.asciidoc

```diff
@@ -3,117 +3,367 @@
+// TEST[s/typed_keys/typed_keys&size=0/]
+// TEST[s/my-field/http.response.bytes/]
+
+The response returns the aggregation type as a prefix to the aggregation's name.
+
+IMPORTANT: Some aggregations return a different aggregation type from the
+type in the request. For example, the terms,
+<<search-aggregations-bucket-significantterms-aggregation,significant terms>>,
+and <<search-aggregations-metrics-percentile-aggregation,percentiles>>
+aggregations return different aggregations types depending on the data type of
+the aggregated field.
+
+[source,console-result]
+----
+{
+  ...
+  "aggregations": {
+    "histogram#my-agg-name": {                 <1>
+      "buckets": []
+    }
+  }
+}
+----
+// TESTRESPONSE[s/\.\.\./"took": "$body.took", "timed_out": false, "_shards": "$body._shards", "hits": "$body.hits",/]
+// TESTRESPONSE[s/"buckets": \[\]/"buckets":\[\{"key":1070000.0,"doc_count":5\}\]/]
+
+<1> The aggregation type, `histogram`, followed by a `#` separator and the aggregation's name, `my-agg-name`.
+
+[discrete]
+[[use-scripts-in-an-agg]]
+=== Use scripts in an aggregation
+
+Some aggregations support <<modules-scripting,scripts>>. You can
+use a `script` to extract or generate values for the aggregation:
+
+[source,console]
+----
+GET /my-index-000001/_search
+{
+  "aggs": {
+    "my-agg-name": {
+      "histogram": {
+        "interval": 1000,
+        "script": {
+          "source": "doc['my-field'].value.length()"
+        }
+      }
+    }
+  }
+}
+----
+// TEST[setup:my_index]
+// TEST[s/my-field/http.request.method/]
+
+If you also specify a `field`, the `script` modifies the field values used in
+the aggregation. The following aggregation uses a script to modify `my-field`
+values:
+
+[source,console]
+----
+GET /my-index-000001/_search
+{
+  "aggs": {
+    "my-agg-name": {
+      "histogram": {
+        "field": "my-field",
+        "interval": 1000,
+        "script": "_value / 1000"
+      }
+    }
+  }
+}
+----
+// TEST[setup:my_index]
+// TEST[s/my-field/http.response.bytes/]
+
+Some aggregations only work on specific data types. Use the `value_type`
+parameter to specify a data type for a script-generated value or an unmapped
+field. `value_type` accepts the following values:
+
+* `boolean`
+* `date`
+* `double`, used for all floating-point numbers
+* `long`, used for all integers
+* `ip`
+* `string`
+
+[source,console]
+----
+GET /my-index-000001/_search
+{
+  "aggs": {
+    "my-agg-name": {
+      "histogram": {
+        "field": "my-field",
+        "interval": 1000,
+        "script": "_value / 1000",
+        "value_type": "long"
+      }
+    }
+  }
+}
+----
+// TEST[setup:my_index]
+// TEST[s/my-field/http.response.bytes/]
+
+[discrete]
+[[agg-caches]]
+=== Aggregation caches
+
+For faster responses, {es} caches the results of frequently run aggregations in
+the <<shard-request-cache,shard request cache>>. To get cached results, use the
+same <<shard-and-node-preference,`preference` string>> for each search. {es}
```

Comment:
I think that the `preference` is less important than making sure you have `size:0`. If you have two shard copies you'll fill the cache pretty quick without preference. But if you don't have `size:0` you can't get it at all!

Reply:
I updated this recommending setting `size` to `0` if you don't need search results.